### PR TITLE
test(setup-node): add shellcheck to nodes

### PR DESCRIPTION
**Pull request by Joshua-Anderson**

### CODE REVIEW

**Comment by mboersma** [comment]
Code LGTM.

File: tests/bin/setup-node.sh [hunk 1]
```diff
@@ -37,6 +37,15 @@ apt-get install -yq curl mercurial python-dev libffi-dev libpq-dev libyaml-dev g
...
 pip install virtualenv
 
+# Use cabal (Haskell installer) to build and install ShellCheck
+apt-get install -yq cabal-install
+cabal update
+cd /tmp
+git clone --single-branch https://github.com/koalaman/shellcheck.git
+cd shellcheck
+cabal install --global
+apt-get purge cabal-install
```
Comment (bacongobbler) [inline tests/bin/setup-node.sh]: we should add in the -y flag here as well just to make sure this won't break :)

File: tests/bin/setup-node.sh [hunk 2]
```diff
@@ -37,6 +37,15 @@ apt-get install -yq curl mercurial python-dev libffi-dev libpq-dev libyaml-dev g
 curl -sSL https://raw.githubusercontent.com/pypa/pip/7.0.3/contrib/get-pip.py | python -
 pip install virtualenv
 
+# Use cabal (Haskell installer) to build and install ShellCheck
+apt-get install -yq cabal-install
+cabal update
+cd /tmp
+git clone --single-branch https://github.com/koalaman/shellcheck.git
```
Comment (bacongobbler) [inline tests/bin/setup-node.sh]: Shouldn't we be cloning a more deterministic branch like v0.3.8?

**Comment by Joshua-Anderson** [comment]
Deferring to @mboersma on the feedback since he already ran these commands on the Jenkins nodes.

**Comment by mboersma** [comment]
@bacongobbler is right, both those changes would be preferable.

**Comment by Joshua-Anderson** [comment]
Done!

**Comment by mboersma** [comment]
Code LGTM++.

**Comment by bacongobbler** [comment]
Lgtm

**Pull request merged**